src/main/java/com/groupesan/project/java/scrumsimulator/mainpackage/ui/widgets/BlockerWidget.java

```diff
@@ -129,15 +139,24 @@ private void linkUserStoryToBlocker() {
     }
 
     private void resolveBlocker() {
-        int confirmation = JOptionPane.showConfirmDialog(
+        List<Solution> solutions = SolutionStore.getInstance(simulationId).getSolutions();
+        Solution selectedSolution = (Solution) JOptionPane.showInputDialog(
                 this,
-                "Are you sure you want to resolve this blocker?",
-                "Resolve Confirmation",
-                JOptionPane.YES_NO_OPTION);
+                "Select a solution to resolve the blocker:",
+                "Resolve Blocker",
+                JOptionPane.PLAIN_MESSAGE,
+                null,
+                solutions.toArray(),
+                null);
+
+        if (selectedSolution != null) {
+            blocker.resolve(selectedSolution);
+
+            solutionLabel.setText("Solution: " + selectedSolution.getTitle());
+            resolveButton.setText("Resolved");
+            resolveButton.setEnabled(false);
```

Comment:
Simultaneously after this step, please store the solutionId (Chosen for resolving blocker) and blockerId( Resolved with solution), add under respective blocker and solution in simulation.json as we discussed during design.

Reply:
OKay, I'll create a new issue and start working on this functionality.